### PR TITLE
RegisterEhrid function

### PR DESCRIPTION
**Pull request by jendakol**
RegisterEhrid function
\+ many util packages and functions

### CODE REVIEW

File: internal/register-ehrid/register-ehrid.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+		var registration = structs.Registration(request)
+		logger.Debugf("Generated new eHrid %v, saving registration %v", ehrid, registration)
+
+		return tx.Set(doc, registration)
+	})
+
+	if err != nil {
+		if err.Error() == NeedsRetry {
+			goto tryNextEhrid
```
Comment (marekaf) [inline internal/register-ehrid/register-ehrid.go]: could we get rid of the goto? :D

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package constants
+
+const FirebaseUrl = "https://erouska-key-server-dev.firebaseio.com/"
```
Comment (marekaf) [inline pkg/constants/constants.go]: tyhle veci bych urcite delal konfigurovatelny pres ENV promenny, zatim to tam nech, ale pripadne pak udelam PR a predelam to :)

File: pkg/httpserver/httpserver.go [hunk 3]
```diff
@@ -5,13 +5,13 @@ import (
...
 	"errors"
 	"fmt"
+	"github.com/covid19cz/erouska-backend/pkg/logging"
 	"net"
 	"net/http"
 	"strconv"
 	"time"
 
 	"github.com/covid19cz/erouska-backend/internal/hello"
-	"github.com/covid19cz/erouska-backend/internal/logging"
```
Comment (marekaf) [inline pkg/httpserver/httpserver.go:14]: @jendakol proc se tohle presunulo z internal do pkg? Mi moc nedava smysl mit nas custom logging exposnutej do pkg, kde to pak muze vybizet ostatni aby si ho naimportovali do svych projektu.

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	"math/rand"
+	"time"
+)
+
+var SeededRand *rand.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+var Validate *validator.Validate
+
+func init() {
+	Validate = validator.New()
```
Comment (marekaf) [inline pkg/utils/utils.go]: what is this for?

Comment (jendakol) [reply pkg/utils/utils.go]: Validation of incoming requests. I've found out that unmarshalling JSON into struct will use default values where the proper field is not available - that seems :scream: to me but fortunately, there's a way how to validate the struct doesn't contain default values.  
Pretty powerful package, BTW.

**Pull request merged**